Examples/ObjectiveC/CacheManagementExample.m

```diff
@@ -0,0 +1,158 @@
+#import "CacheManagementExample.h"
+@import Mapbox;
+
+NSString *const MBXExampleCacheManagement = @"CacheManagementExample";
+
+@interface CacheManagementExample () <MGLMapViewDelegate>
+
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) UIAlertController *alertController;
+@end
+
+@implementation CacheManagementExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    /* Set the maximum ambient cache size in bytes. Call this method before the map view is loaded.
+
+     The ambient cache is created through the end user loading and using a map view. */
+    NSUInteger maxCacheSize = 62914560;
+    [[MGLOfflineStorage sharedOfflineStorage] setMaximumAmbientCacheSize:maxCacheSize withCompletionHandler:^(NSError * _Nullable error) {
+        if (error != nil) {
+            NSLog(@"Unable to set maximum ambient cache size: %@", error.localizedDescription);
+        }
+
+        self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds];
+        self.mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+        self.mapView.delegate = self;
+
+        [self.view addSubview:self.mapView];
+
+        // Create an offline pack.
+        [self addOfflinePack];
+    }];
+
+    // Add a bar button. Tapping this button will present a menu of options. For this example, the cache is managed through the UI. It can also be managed by developers through remote notifications.
+    // For more information about managing remote notifications in your iOS app, see the Apple "UserNotifications" documentation: https://developer.apple.com/documentation/usernotifications?language=objc
+    UIBarButtonItem *alertButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemEdit target:self action:@selector(presentActionSheet)];
+    [self.parentViewController.navigationItem setRightBarButtonItem:alertButton];
+}
+
+#pragma mark: Cache management methods called by action sheet
+
+// Check whether the tiles locally cached match those on the tile server. If the local tiles are out-of-date, they will be updated. Invalidating the ambient cache is preferred to clearing the cache. Tiles shared with offline packs will not be affected by this method.
+- (void)invalidateAmbientCache {
+    CFTimeInterval start = CACurrentMediaTime();
+    [[MGLOfflineStorage sharedOfflineStorage] invalidateAmbientCacheWithCompletionHandler:^(NSError * _Nullable error) {
+        if (error) {
+            NSLog(@"Error: %@", error.localizedDescription);
+            return;
+        } else {
+            CFTimeInterval difference = CACurrentMediaTime() - start;
+
+            // Display an alert to indicate that the invalidation is complete.
+            [self presentCompletionAlertWithTitle:@"Ambient Cache Invalidated" andMessage: [NSString stringWithFormat: @"Invalidated ambient cache in %f seconds", difference]];
```

Comment:
References to `self` inside blocks should typically be `weak` or `unowned`. Additionally, UI work must happen on the main thread, so does this?